Packs/BarracudaEmailProtection/ModelingRules/BarracudaEmailProtection_1_3/BarracudaEmailProtection_1_3.xif

```diff
@@ -0,0 +1,18 @@
+[MODEL:dataset="barracuda_email_protection_raw"]
+alter 
+    payload = arrayindex(regextract(_raw_log, "\[\d\]: (\{.*?$)"), 0) |
+alter
+     xdm.target.domain = json_extract_scalar(payload, "$.dst_domain")
+    ,xdm.target.user.username = json_extract_scalar(payload, "$.hdr_to")
+    ,xdm.email.sender = arrayindex(regextract(json_extract_scalar(payload, "$.hdr_from"), "\<(.*?)\>"), 0) 
+    ,xdm.email.subject = json_extract_scalar(payload, "$.subject")
+    ,xdm.email.message_id = json_extract_scalar(payload, "$.message_id")
+    ,xdm.email.delivery_timestamp = to_timestamp(parse_epoch("%Y-%m-%dT%H:%M:%S%z", json_extract_scalar(payload, "$.timestamp")) , "SECONDS")
+    ,xdm.email.recipients = arraymap(json_extract_array (payload,"$.recipients"), json_extract_scalar("@element", "$.email"))
+    ,xdm.source.sent_bytes = to_integer(json_extract(payload, "$.size"))
+    ,xdm.source.ipv4 = json_extract_scalar(payload, "$.src_ip")
+    ,xdm.source.user.username = arrayindex(regextract(json_extract_scalar(payload, "$.hdr_from"), "\"(.*?)\""), 0)
+    # TODO add mapping
+    # ,xdm.observer.action => recipients.action
+    # ,xdm.event.outcome => recipients.delivered
```

Comment:
@eepstain @evisochek - In the demo there was a request to map these two fields. However, I think there might a problem with the event schema that wouldn't allow us to map these two. 

The `delivered` and `action` fields are inside the `recipients` array of objects. So for example, if we get multiple recipients such as:

```json
{
	"recipients": [
	   {
	     "action":"allowed",
	     "delivered":"delivered",
	     "email":"twellick@bfgnetworks.com"
	   },
	   {
	     "action":"not allowed",
	     "delivered":"not delivered",
	     "email":"bwellick@bfgnetworks.com"
	   },
	  // ...
	],
}
```

We can see that `action` and `delivered` can have multiple values, allowed/not allowed and delivered/not delivered, respectively.
In this case, we can't map `xdm.event.outcome` to both delivered and not delivered. Same for `xdm.observer.action`. 

Please advise.